geonode/geoserver/geofence.py

```diff
@@ -310,73 +340,127 @@ def insert_rule(self, rule: Rule):
             http://<host>:<port>/geoserver/rest/geofence/rules
             """
             r = requests.post(
-                f'{self.baseurl.rstrip("/")}/geofence/rules',
-                # headers={'Content-type': 'application/json'},
+                f"{self.baseurl}rules",
                 json=rule.get_object(),
                 auth=HTTPBasicAuth(self.username, self.pw),
-                timeout=ogc_server_settings.get("TIMEOUT", 60),
+                timeout=self.timeout,
                 verify=False,
             )
 
             if r.status_code not in (200, 201):
-                logger.debug(f"Could not insert rule: [{r.status_code}] - {r.content}")
-                raise GeofenceException(f"Could not insert rule: [{r.status_code}]")
+                logger.debug(f"Could not insert rule: [{r.status_code}] - {r.text}")
+                raise GeofenceException(f"Could not insert rule: [{r.status_code}] - {r.text}")
 
         except Exception as e:
             logger.debug("Error while inserting rule", exc_info=e)
             raise GeofenceException(f"Error while inserting rule: {e}")
 
-    def run_batch(self, batch: Batch):
-        if batch.get_batch_length() == 0:
+    def run_batch(self, batch: Batch, timeout: int = None) -> bool:
+        if batch.length() == 0:
             logger.debug(f"Skipping batch execution {batch.log_name}")
-            return
+            return False
 
+        logger.debug(f"Running batch {batch.log_name} with {batch.length()} operations")
         try:
             """
             curl -X GET -u admin:geoserver \
                 http://<host>:<port>/geoserver/rest/geofence/rules/count.json
             """
             r = requests.post(
-                f'{self.baseurl.rstrip("/")}/geofence/batch/exec',
+                f"{self.baseurl}batch/exec",
                 json=batch.get_object(),
                 auth=HTTPBasicAuth(self.username, self.pw),
-                timeout=ogc_server_settings.get("TIMEOUT", 60),
+                timeout=timeout or self.timeout,
                 verify=False,
             )
 
             if r.status_code != 200:
-                logger.debug(f"Error while running batch {batch.log_name}: [{r.status_code}] - {r.content}")
+                logger.debug(
+                    f"Error while running batch {batch.log_name}: [{r.status_code}] - {r.content}"
+                    f"\n {batch.get_object()}"
+                )
                 raise GeofenceException(f"Error while running batch {batch.log_name}: [{r.status_code}]")
 
-            return
+            return True
 
         except Exception as e:
-            logger.debug(f"Error while requesting batch execution {batch.log_name}", exc_info=e)
+            logger.info(f"Error while requesting batch exec {batch.log_name}")
+            logger.debug(f"Error while requesting batch exec {batch.log_name} --> {batch.get_object()}", exc_info=e)
```

Comment:
It might be useful to increase this log level if it shows additional information than the Exception.

Reply:
Usually the detailed information should be logged only when debugging